### PR TITLE
CCR: encode and decode auto-follow pattern name

### PR DESCRIPTION
**Pull request by loixlab**

### CODE REVIEW

File: x-pack/plugins/cross_cluster_replication/public/app/services/routing.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ class Routing {
       : undefined;
 
     this._reactRouter.history.push({
-      pathname: appToBasePathMap[app] + route,
+      pathname: encodeURI(appToBasePathMap[app] + route),
```
Comment (cjcenizal) [inline x-pack/plugins/cross_cluster_replication/public/app/services/routing.js:71]: In which situations is this necessary?

**Review by cjcenizal — APPROVED** [review]
Looks great! Had one question.

File: x-pack/plugins/cross_cluster_replication/public/app/services/routing.js (hunk 1 above)
Comment (loixlab) [reply x-pack/plugins/cross_cluster_replication/public/app/services/routing.js:71]: Whenever we navigate to a route that might contain special characters. Like when editing an auto-follow pattern with this name :`%&()&"` 😊 The url `/auto_follow_patterns/edit/<name>` needs to be encoded

Comment (cjcenizal) [reply x-pack/plugins/cross_cluster_replication/public/app/services/routing.js:71]: Aren't we already encoding the name when we call `navigate`? e.g. https://github.com/sebelga/kibana/pull/5/files#diff-5ec658710f360dfa483d45245036ba62R150

**Pull request closed without merging**